swap/src/bob/swap.rs

```diff
@@ -272,18 +276,27 @@ where
                 .await
             }
             BobState::XmrLocked(state, alice_peer_id) => {
-                // Alice has locked Xmr
-                // Bob sends Alice his key
-                let tx_redeem_encsig = state.tx_redeem_encsig();
-                // Do we have to wait for a response?
-                // What if Alice fails to receive this? Should we always resend?
-                // todo: If we cannot dial Alice we should go to EncSigSent. Maybe dialing
-                // should happen in this arm?
-                event_loop_handle
-                    .send_message3(alice_peer_id.clone(), tx_redeem_encsig)
-                    .await?;
+                let state = if let Epoch::T0 = state.current_epoch(bitcoin_wallet.as_ref()).await? {
+                    // Alice has locked Xmr
+                    // Bob sends Alice his key
+                    let tx_redeem_encsig = state.tx_redeem_encsig();
+
+                    let state4_clone = state.clone();
+                    let enc_sig_sent_watcher =
+                        event_loop_handle.send_message3(alice_peer_id.clone(), tx_redeem_encsig);
+                    let bitcoin_wallet = bitcoin_wallet.clone();
+                    let t1_timeout = state4_clone.wait_for_t1(bitcoin_wallet.as_ref());
+
+                    pin_mut!(enc_sig_sent_watcher);
```

Comment:
Let's prefer `select!` over `pin_mut`.

Reply:
OK, I will change ALL of the `pin_mut` to `select` then.
I created this PR under the assumption that the cleanup would rebase on these changes and then change this. Did not want to mix it... :D

Reply:
Don't over do it, just don't introduce new `pin_mut`.